src/config.rs

```diff
@@ -60,6 +60,33 @@ impl WriteOptions {
     }
 }
 
+/// defines the clock that SlateDB will use during this session
+pub trait Clock {
+    /// Returns a timestamp in milliseconds since the unix epoch,
+    /// must return monotonically increasing numbers (this is enforced
+    /// at runtime and will panic if the invariant is broken)
+    ///
+    /// Note that this clock does not need to return a number that
+    /// represents the unix timestamp; the only requirement is that
+    /// it represents a sequence that can attribute a logical ordering
+    /// to actions on the database
+    fn now(&self) -> i64;
+}
+
+/// contains the default implementation of the Clock, and will return the system time
+pub struct SystemClock {
+    pub system_time: SystemTime,
+}
+
+impl Clock for SystemClock {
+    fn now(&self) -> i64 {
+        match self.system_time.duration_since(UNIX_EPOCH) {
```

Comment:
This was just the way to return a `i64`, the default `SystemTime::now` returns an instance of `SystemTime`

Reply:
Conversed offline, I didn't get how the system time works in Rust. Will fix

Reply:
`or Instant::now since SystemTime is not monotonic` - nvm `Instant` cannot be used to measure epoch times. We'll need to use `SystemTime` but then apply a `min` with the last computed time.